dipy/reconst/ivim.py

```diff
@@ -0,0 +1,478 @@
+        200).Estimate f using 1 - S0_prime/S0. Use least squares to fit D_star
+        and f.
+
+        We do a final fitting of all four parameters after scaling the
+        parameters and select the set of parameters which make sense physically.
+        The criteria for selecting a particular set of parameters is checking
+        the perfusion fraction. If the fraction is more than 30%, we will reject
+        the solution obtained without scaling.
+
+
+        Parameters
+        ----------
+        data : array
+            The measured signal from one voxel. A multi voxel decorator
+            will be applied to this fit method to scale it and apply it
+            to multiple voxels.
+
+        mask : array
+            A boolean array used to mark the coordinates in the data that
+            should be analyzed that has the shape data.shape[:-1]
+
+        Returns
+        -------
+        IvimFit object
+        """
+        # Use the function estimate_S0_prime_D to get S0_prime and D.
+        S0_prime, D = self.estimate_S0_prime_D(data)
+        S0, D_star_prime = self.estimate_S0_D_star_prime(data)
+        f_guess = 1 - S0_prime / S0
+        x0_guess = np.array([S0, f_guess, D_star_prime, D])
+        try:
+            f, D_star = self.estimate_f_D_star(data, x0_guess)
+        except:
+            warningMsg = "x0 obtained from linear fitting is unfeasibile."
+            warningMsg += "Using parameters from the linear fitting"
+            warnings.warn(warningMsg, UserWarning)
+            f, D_star = f_guess, D_star_prime
+
+        x0 = np.array([S0, f, D_star, D])
+        # Fit parameters again with scaling
+        params_with_scaling = self._leastsq(data, x0, scaling=self.scaling)
+        # Fit parameters without scaling
+        params_no_scaling = self._leastsq(data, x0)
+
+        params_in_mask = np.where(params_no_scaling[1] <= .5,
+                                  params_no_scaling, params_with_scaling)
+        return IvimFit(self, params_in_mask)
+
+    def estimate_S0_prime_D(self, data):
+        """Estimate S0_prime and D for bvals > split_b_D
+        """
+        bvals_ge_split = self.gtab.bvals[self.gtab.bvals >= self.split_b_D]
+        bvecs_ge_split = self.gtab.bvecs[self.gtab.bvals >= self.split_b_D]
+        gtab_ge_split = gradient_table(bvals_ge_split, bvecs_ge_split.T)
+
+        D, neg_log_S0 = np.polyfit(gtab_ge_split.bvals,
+                                   -np.log(data[self.gtab.bvals >=
+                                           self.split_b_D]), 1)
+        S0_prime = np.exp(-neg_log_S0)
+        return S0_prime, D
+
+    def estimate_S0_D_star_prime(self, data):
+        """Estimate S0 and D_star_prime for bvals < split_b_S0
+        """
+        bvals_le_split = self.gtab.bvals[self.gtab.bvals < self.split_b_S0]
+        bvecs_le_split = self.gtab.bvecs[self.gtab.bvals < self.split_b_S0]
+        gtab_le_split = gradient_table(bvals_le_split, bvecs_le_split.T)
```

Comment:
I don't think you need the gtab here. These three lines could be changed to:

    bvals_le_split = self.gtab.bvals[self.gtab.bvals < self.split_b_S0]

And then line 283 would change to: 

    D_star_prime, neg_log_S0 = np.polyfit(bvals_le_split,